### PR TITLE
Handle conversion of constants in parser

### PR DESCRIPTION
**Pull request by nairr**
Adds conversion factors for AQUCT constants to SI units

### CODE REVIEW

**Comment by joakim-hove** [comment]
This looks like an extra bolted on unit system which I do not understand; it will not be merged in the current form. Before I can suggest an alternative implementation I must understand better what it is supposed to achieve:

1. Why do do need the `p0_defaulted` flag - if possible should defaulted or not behaviour should be accounted for at the level of the current code, and not held on to/passed further.

2. Where are the conversion factors `c1` and `c2`used? Hopefully the required conversions can be performed - without the need to hold on to/pass on the `c1` and `c2` values?

**Comment by atgeirr** [comment]
I'd add: what are the units here? If at all possible (i.e. the constant conversion does not involve completely arbitrary numbers but numbers that are combination of other unit conversion factors) it should be simple to use the existing unit system, or extend it if really necessary.

**Comment by nairr** [comment]
@joakim-hove - 1. My mistake, I've now used p0_defaulted within aquiferct.   2.  The conversion factors are used in AquiferCarterTracy.hpp, lines 293 and 299. I'm not aware how to do this conversion in a different way. Please let me know if you have some insights

**Comment by joakim-hove** [comment]
> 2. The conversion factors are used in AquiferCarterTracy.hpp, lines 293 and 299. I'm not aware how to do this conversion in a different way. Please let me know if you have some insights

Sorry - I can not find this location; please provide a link.

**Comment by nairr** [comment]
ok, will do. The file is located here: https://github.com/kel85uk/opm-simulators/blob/e421a8eca59abbdaa9ec1dffce690b06189d9b15/opm/autodiff/AquiferCarterTracy.hpp

**Comment by nairr** [comment]
:) sorry for sending you on a wild goose chase!

File: src/opm/parser/eclipse/EclipseState/AquiferCT.cpp [hunk 1]
```diff
@@ -31,17 +31,26 @@ namespace Opm {
         for (auto& aquctRecord : aquctKeyword){
   
             AquiferCT::AQUCT_data data;
-    
+          
+            if( deck.hasKeyword( "FIELD" ) ) {
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/EclipseState/AquiferCT.cpp]: OK; the best solution here would be to invoke the unit system to perform a numerical conversion on your behalf, but I realize that will require some code changes in the unit system. This should be an OK solution:

const auto& unitSystem = deck.getUnitSystem();
switch( unitSystem.getType() ) {
     case FIELD:
         c1 = ..
         c2 = ...
         break;
      case METRIC:
         c1 = 
         c2 = 
         break
     LAB:
     
..

**Comment by atgeirr** [comment]
This is still open, we should try to have it completed. Is it not possible to describe the constants in a unit-invariant/SI way, i.e. 2*pi for c2, and then multiply by whatever is needed to convert, using the opm unit system?

It also looks to me like the values of c1 and c2 should not depend on any unit at all: there is no input to convert! The weird calculation for c2 (`data.c2 = 1.1191*5.6164`) just gives 2*pi again. And if c1 varies depending on the unit system, I cannot see how that can produce correct results: if I describe the same reservoir in both field and metric units (in two different decks), I would expect the internal representation inside the simulator to be very close (up to roundoff), and using different values for c1 would produce different results (in SI, internally)!

**Comment by nairr** [comment]
Sorry for the delay in closing this. Was out of office. 
@atgeirr - you are absolutely right. Once all the conversions are carried out internally,  c2 is basically 2*pi and  c1 = 1     (0.00827*117.26852  or 0.006328*1.5797e2)

I've made the required changes. This should be mergeable now.

**Comment by nairr** [comment]
correction : (0.008527*117.26852 or 0.006328*1.5797e2) both the conversions of C1 are approximately equal to one.

**Comment by atgeirr** [comment]
Then the code should document that with a comment explaining that this is the value used by Eclipse, so we use it also, to produce equivalent results.

File: src/opm/parser/eclipse/EclipseState/AquiferCT.cpp [hunk 2]
```diff
@@ -31,17 +31,18 @@ namespace Opm {
         for (auto& aquctRecord : aquctKeyword){
   
             AquiferCT::AQUCT_data data;
-    
+
+            data.c1 = 1.0;
+            data.c2 = 6.283;        // Value of C2 used by E100 (for METRIC, PVT-M and LAB unit systems)
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/EclipseState/AquiferCT.cpp]: This looks suspiciosly like 2&pi;?

**Comment by nairr** [comment]
Yes, it does look to be 2 * pi, but I don't have a justification for it and E100 reports that it uses the value 6.283 for c2. 

I leave the choice of whether to use 2 * pi or 6.283 up to you. The change in results are really negligible for my test case.

**Pull request merged**